### PR TITLE
Feature/standard context api

### PR DESCRIPTION
**Pull request by EstebanDalelR**
## Description
Uses new standard API, preventing errors
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  
## Acceptance

- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by baristaGeek** [comment]
It seems to work but I do want to say that there's a caveat in my testing strategy: I'm logged out of the services and for some reason can't login via the UI (must be some edge case due to being a tester).

File: src/extension.ts [hunk 1]
```diff
@@ -99,127 +85,37 @@ export class WatermelonTreeDataProvider
...
+      uniqueBlames = await getBlame(gitAPI, startLine, endLine);
+    }
 
-      let issuesWithTitlesAndGroupedComments = await getPRsToPaintPerSHAs({
-        arrayOfSHAs,
+    if (!session) {
+      setLoggedIn(false);
+      return items;
+    }
+    const [contextItems] = await Promise.all([
```
Comment (baristaGeek) [inline src/extension.ts:107]: Nice! This also looks syntactically way more organized. Question: We depend on a lot of calls to different APIs and we want to show the ones that succeed even if one fails (eg: Slack is down, Open AI is down). Would promise.allSettled( ) suffice this better?

Just a minor detail that regardless of the answer, gives more context.

**Review by baristaGeek — COMMENTED** [review]
LGTM, but I'm a little bit resistant to approve it because of this behavior. My GH username exists on the DB. 

https://github.com/watermelontools/watermelon-vscode/assets/8325094/ac4f2d63-b01c-4f2b-9ffb-b4f49dec212d

**Comment by EstebanDalelR** [comment]
> It seems to work but I do want to say that there's a caveat in my testing strategy: I'm logged out of the services and for some reason can't login via the UI (must be some edge case due to being a tester).

Please explain better?

File: src/extension.ts (hunk 1 above)
Comment (EstebanDalelR) [reply src/extension.ts:107]: For now, we show only the ones that return properly. We could make the UI show the errors that the backend sends.

**Comment by EstebanDalelR** [comment]
> LGTM, but I'm a little bit resistant to approve it because of this behavior. My GH username exists on the DB.
> Screen.Recording.2023-08-25.at.3.56.07.PM.mov

What failed?

**Comment by baristaGeek** [comment]
> > It seems to work but I do want to say that there's a caveat in my testing strategy: I'm logged out of the services and for some reason can't login via the UI (must be some edge case due to being a tester).
> 
> Please explain better?

I was running on turborepo and got a `no git system` error (I was pointing to the prod API). Later ran on watermelon's backend repo and worked (pointing to prod). Later tried again on turborepo pointing to localhost and worked. 

Weird behavior to take notes of, but let's merge.

**Review by baristaGeek — APPROVED** [review]

**Pull request merged**